### ISSUE TITLE
Segmentation fault on incorrect case for parameter name

### ISSUE THREAD

**Issue by sergeyklay** [daveshanley/vacuum#240]
Hi,

The following code:

```yaml
# ...

components:
  parameters:
    CategoryFilter:
      in: query
      name: category
      description: Filter by product category.
      required: false
      schema:
        type: string
      examples:
        single:
          value: laptops
          summary: A sample category to filter products.

paths:
  /v1/products:
    get:
      summary: Get products
      description: Returns a list of all products.
      operationId: products.index
      tags: [products]
      parameters:
        - $ref: '#/components/parameters/categoryFilter'

# ...
```

leads to segmentation fault as show bellow:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x20 pc=0x10134f3d0]

goroutine 59 [running]:
github.com/daveshanley/vacuum/functions/openapi.OperationParameters.RunRule({}, {0x14000124048?, 0x8?, 0x14000088000?}, {0x14000114150, 0x140001048f0, {0x10196de00, 0x101a4b378}, {0x0, 0x0}, ...})
	/Users/egrep/go/pkg/mod/github.com/daveshanley/vacuum@v0.0.47/functions/openapi/operation_parameters.go:73 +0xa00
github.com/daveshanley/vacuum/motor.buildResults({0x140001048f0, 0x140003e8aa0, {0x101a50670, 0x140003e0410}, 0x140000bff08, 0x140003caa00, 0x140004960c0, 0x14000539800, 0x140003e8b40, 0x0, ...}, ...)
	/Users/egrep/go/pkg/mod/github.com/daveshanley/vacuum@v0.0.47/motor/rule_applicator.go:400 +0x55c
github.com/daveshanley/vacuum/motor.runRule({0x140001048f0, 0x140003e8aa0, {0x101a50670, 0x140003e0410}, 0x140000bff08, 0x140003caa00, 0x140004960c0, 0x14000539800, 0x140003e8b40, 0x0, ...})
	/Users/egrep/go/pkg/mod/github.com/daveshanley/vacuum@v0.0.47/motor/rule_applicator.go:327 +0x36c
created by github.com/daveshanley/vacuum/motor.ApplyRulesToRuleSet
	/Users/egrep/go/pkg/mod/github.com/daveshanley/vacuum@v0.0.47/motor/rule_applicator.go:162 +0x628
```

Of course this can be fixed in the following way:
```diff
# ...

components:
  parameters:
    CategoryFilter:
      in: query
      name: category
      description: Filter by product category.
      required: false
      schema:
        type: string
      examples:
        single:
          value: laptops
          summary: A sample category to filter products.

paths:
  /v1/products:
    get:
      summary: Get products
      description: Returns a list of all products.
      operationId: products.index
      tags: [products]
      parameters:
-       - $ref: '#/components/parameters/categoryFilter'
+       - $ref: '#/components/parameters/CategoryFilter'

# ...

```

However, finding this error in automatic mode at CI/CD phase is almost impossible because vacuum crashes without specifying on which line you have an error.

Steps to reproduce: https://github.com/sergeyklay/specmatic-testing-example/pull/8 (see "**Lint OpenAPI**" check)

**Comment by daveshanley** [comment]
Nice find, I will look into this for you ASAP.

**Comment by daveshanley** [comment]
Resolved in `v0.0.52`

**Comment by sergeyklay** [comment]
Thank you!
